src/nano_contracts/builder.ts

```diff
@@ -324,47 +441,104 @@ class NanoContractTransactionBuilder {
           outputs = concat(outputs, ret[1]);
         } else if (action.type === NanoContractActionType.WITHDRAWAL) {
           const output = this.executeWithdrawal(action, tokens);
-          outputs = concat(outputs, output);
+          if (output) {
+            outputs = concat(outputs, output);
+          }
         } else {
           throw new Error('Invalid type for nano contract action.');
         }
       }
     }
 
-    // Serialize the method arguments
-    const serializedArgs: Buffer[] = [];
-    if (this.args) {
-      const serializer = new Serializer();
-      const blueprintInformation = await ncApi.getBlueprintInformation(this.blueprintId);
-      const methodArgs = get(
-        blueprintInformation,
-        `public_methods.${this.method}.args`,
-        []
-      ) as MethodArgInfo[];
-      if (!methodArgs) {
-        throw new NanoContractTransactionError(`Blueprint does not have method ${this.method}.`);
-      }
+    return [inputs, outputs, tokens];
+  }
 
-      if (this.args.length !== methodArgs.length) {
-        throw new NanoContractTransactionError(
-          `Method needs ${methodArgs.length} parameters but data has ${this.args.length}.`
-        );
-      }
+  /**
+   * Build a transaction object from the built inputs/outputs/tokens
+   *
+   * It will create a Transaction or CreateTokenTransaction, depending on the vertex type
+   *
+   * @throws {NanoContractTransactionError} In case the create token options is null
+   *
+   * @memberof NanoContractTransactionBuilder
+   * @inner
+   */
+  async buildTransaction(
+    inputs: IDataInput[],
+    outputs: IDataOutput[],
+    tokens: string[]
+  ): Promise<Transaction | CreateTokenTransaction> {
+    if (this.vertexType === NanoContractVertexType.TRANSACTION) {
+      return transactionUtils.createTransactionFromData(
+        {
+          version: DEFAULT_TX_VERSION,
+          inputs,
+          outputs,
+          tokens,
+        },
+        this.wallet.getNetworkObject()
+      );
+    }
 
-      for (const [index, arg] of methodArgs.entries()) {
-        const serialized = serializer.serializeFromType(this.args[index], arg.type);
-        serializedArgs.push(serialized);
-      }
+    if (this.createTokenOptions === null) {
+      throw new NanoContractTransactionError(
+        "Create token options can't be null when creating a create token transaction."
+      );
     }
 
+    // It's a token creation transaction
+    // then we get the token creation data from the utils method
+    // and concatenate the nano actions inputs/outputs/tokens
+    const data = await tokensUtils.prepareCreateTokenData(
```

Comment:
Maybe we should have this inside an if and not the "default".
If the vertex type is not `TRANSACTION` or `CREATE_TOKEN_TRANSACTION` we throw an error.
What do you think?

Reply:
Makes sense! https://github.com/HathorNetwork/hathor-wallet-lib/pull/860/commits/c98a7eb481d8ea3f7dff158a0b65d85e307ecf95